src/coreclr/src/pal/src/thread/process.cpp

```diff
@@ -3365,7 +3385,12 @@ InitializeFlushProcessWriteBuffers()
 
     if (status != 0)
     {
+#if defined(TARGET_OSX) && defined(TARGET_AMD64)
+        s_helperPage = 0;
```

Comment:
We also should unmap the allocated page. We didn't have to do it before as we were going to abort the process anyways. But with the fallback, the page would just stay mapped and unused.

Reply:
I'll clean it up in the osx-x64 version of the PR.